### ISSUE TITLE
Thyra -Werror build broken

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#53]
@mhoemmen, I believe your commit:

```
89a3771 "Thyra::TpetraVectorSpace: Get rid of unnecessary private fields."
Author: Mark Hoemmen <mhoemme@sandia.gov>
Date:   Sat Dec 5 22:43:22 2015 -0700 (5 days ago)

M       packages/thyra/adapters/tpetra/src/Thyra_TpetraVectorSpace_decl.hpp
M       packages/thyra/adapters/tpetra/src/Thyra_TpetraVectorSpace_def.hpp
```
caused these -Werror build errors shown in the below email.  Can you please fix this?

Are you on the thyra-regression mail list?

-Ross

> -----Original Message-----
> From: thyra-regression-bounces@software.sandia.gov [mailto:thyra-
> regression-bounces@software.sandia.gov] On Behalf Of CDash
> Sent: Friday, December 11, 2015 12:11 AM
> To: thyra-regression@software.sandia.gov
> Subject: [Thyra-Regression] FAILED (b=4): Trilinos/Thyra - Linux-GCC-4.7.2-
> MPI_DEBUG_Werror_DEV - Nightly
> 
> A submission to CDash for the project Trilinos has build errors.
> You have been identified as one of the authors who have checked in changes
> that are part of this submission or you are listed in the default contact list.
> 
> Details on the submission can be found at
> http://testing.sandia.gov/cdash/buildSummary.php?buildid=2264582
> 
> Project: Trilinos
> SubProject: Thyra
> Site: muir.sandia.gov
> Build Name: Linux-GCC-4.7.2-MPI_DEBUG_Werror_DEV
> Build Time: 2015-12-10T22:05:57 MST
> Type: Nightly
> Errors: 4
> 
> 
> *Error*
> packages/thyra/adapters/tpetra/test/TpetraThyraWrappers_UnitTests.cpp
> (http://testing.sandia.gov/cdash/viewBuildError.php?type=0&buildid=22645
> 82)
> In file included from /nightly/hudson/slave/workspace/trilinos-nightly-
> muir/MPI_DEBUG_Werror_DEV/Trilinos/pacg++: error:
> CMakeFiles/ThyraTpetraAdapters_Simple2DTpetraModelEvaluatorUnitTests.
> dir/Simple2DTpetraModelEvaluator_UnitTests.cpp.o: No such file or
> directory
> packages/thyra/adapters/tpetra/test/Simple2DTpetraModelEvaluator_UnitT
> ests.cpp
> (http://testing.sandia.gov/cdash/viewBuildError.php?type=0&buildid=22645
> 82)
> In file included from /nightly/hudson/slave/workspace/trilinos-nightly-
> muir/MPI_DEBUG_Werror_DEV/Trig++: error:
> CMakeFiles/ThyraTpetraAdapters_TpetraThyraWrappersUnitTests.dir/Tpetr
> aThyraWrappers_UnitTests.cpp.o: No such file or directory
> 
> 
> -CDash on testing.sandia.gov
> 
> _______________________________________________
> Thyra-Regression mailing list
> Thyra-Regression@software.sandia.gov
> https://software.sandia.gov/mailman/listinfo/thyra-regression

**Comment by mhoemmen** [comment]
Just curious -- how important are shadowing warnings in tests?  I haven't gotten any complaints about this from the local users who would normally complain.  Is this a show-stopper or can it sit for a little while?  Of course we want to fix all bugs :-) but I'm just looking for a way to set priorities.

**Comment by bartlettroscoe** [comment]
Warnings in tests obscure warnings in your library code when you develop.  You need to have zero warnings.  Why not?

Also, spamming people with CDash failures just conditions people to ignoring CDash emails.

Solve this how you like but please don't let CDash failures keep happening day after day.

**Comment by tawiesn** [comment]
Mark, i just sent you a patch for fixing the shadow warnings that i have laying around for a couple of days. So, i didn't see it any more in my local build. Forgot to push though (i'm quite busy with the Xpetra stuff working on different other branches)...

**Comment by mhoemmen** [comment]
@tawiesn Thanks Tobias!  Would you like me to push it or should I wait for it to go through when you push the Xpetra stuff?

**Comment by mhoemmen** [comment]
@tawiesn Oops, just got your e-mail; it says "feel free to push it."  Thanks!

**Comment by tawiesn** [comment]
Feel free to push it, since the Xpetra stuff will take some more time. I have to rewrite the local history of my patches any way, so i just remove it from there.

**Comment by tawiesn** [comment]
Pushed it. Thyra builds on muir again.

**Comment by bartlettroscoe** [comment]
@tawiesn,

Thanks for taking care of this!

FYI: If you put "Fixes #53" in the fixing commit, this will automatically close the ticket.  If you are not sure if it fixes it for sure, just list "Should address #53" or something in the commit message.  That way, it will show up in the GitHub Issue #53 and make it easier to find.

**Comment by tawiesn** [comment]
Understood.

Tobias
